jina/optimizers/discovery.py

```diff
@@ -0,0 +1,128 @@
+import os
+import shutil
+from pathlib import Path
+
+from jina.executors import BaseExecutor
+from jina.logging import default_logger as logger
+from jina.helper import yaml
+
+
+def _read_file(filename):
+    with open(filename, "r") as file:
+        return file.readlines()
+
+
+def _extract_executor_files(flows):
+    executor_files = set()
+    for flow_definition in flows.values():
+        for line in flow_definition:
+            stripped = line.strip()
+            if stripped.startswith("uses"):
```

Comment:
what happens if I have a `docker image`?

Reply:
Good catch! Any suggestions for changing the logic?

Reply:
We can check for stripped.endswith('.yml')